compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/IrInlineCodegen.kt

```diff
@@ -157,7 +157,8 @@ class IrInlineCodegen(
         try {
             performInline(
                 expression.symbol.owner.typeParameters.map { it.symbol },
-                function.origin == IrDeclarationOrigin.FUNCTION_FOR_DEFAULT_PARAMETER,
+                // Always look for default lambdas to allow custom default argument handling in compiler plugins.
```

Comment:
Real change: In order to allow compiler plugins to have default argument handling in methods that are not FUNCTION_FOR_DEFAULT_PARAMETER, we always look for default lambdas (and bail out if there is no mask below, so there should be no compile-time performance penalty).

Reply:
(This was extracted and merged separately in #3707.)